dimod/generators/bpsp.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2024 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import dimod
+import numpy.random as rnd
+
+__all__ = ["random_binary_paint_shop_problem"]
+
+
+def binary_paint_shop_problem(car_sequence):
+    """
+    Ising formulation for Binary Paint Shop Problem:
+    - Streif, Michael, et al.
+      "Beating classical heuristics for the binary paint shop
+      problem with the quantum approximate optimization algorithm."
+      Physical Review A 104.1 (2021): 012403.
+
+    Input:
+    - car_sequence: list of car labels
+    - seed: Seed for the random number generator
+
+    Returns:
+    - BQM: Ising model where each variable is a car, and the value indicates what color it gets painted first
```

Comment:
For inputs/outputs we follow [Google's Python style guide](https://google.github.io/styleguide/pyguide.html#244-decision) so
```suggestion
    Args:
        car_sequence: A list of car labels.
        seed: Seed for the random number generator.

    Returns:
        Ising model where each variable is a car, and the value indicates what color it gets painted first.
```
here and elsewhere.